File: 7-reverse-integer/reverse-integer.py
class Solution(object):
    def reverse(self, x):
        sign = -1 if x < 0 else 1
        s = str(abs(x))[::-1]
        res = sign * int(s)

        if res < -2**31 or res > 2**31 - 1:
            return 0
        return res

File: 7-reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int res = 0;
        int der;
        while(x != 0){
            der = x % 10;
            x /= 10;
            if (res > INT_MAX / 10 ||
                (res == INT_MAX / 10 && der > INT_MAX % 10)) {
                return 0;
            }

            if (res < INT_MIN / 10 ||
                (res == INT_MIN / 10 && der < INT_MIN % 10)) {
                return 0;
            }            
            res = res * 10 + der;
        }
        return res;       
    }
};